SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp

```diff
@@ -297,78 +298,30 @@ bool FSpatialGDKEditorModule::ForEveryServerWorker(TFunction<void(const FName&,
 	return false;
 }
 
-FPlayInEditorSettingsOverride FSpatialGDKEditorModule::GetPlayInEditorSettingsOverrideForTesting(UWorld* World) const
+void FSpatialGDKEditorModule::OverrideSettingsForTesting(UWorld* World, const FString& MapName)
 {
 	// By default, clear that the runtime/test was loaded from a snapshot taken for a given world.
 	ASpatialFunctionalTest::ClearLoadedFromTakenSnapshot();
 
-	FPlayInEditorSettingsOverride PIESettingsOverride = ISpatialGDKEditorModule::GetPlayInEditorSettingsOverrideForTesting(World);
-	if (const ASpatialWorldSettings* SpatialWorldSettings = Cast<ASpatialWorldSettings>(World->GetWorldSettings()))
+	SpatialTestSettings->Override(MapName);
+
+	if (GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking())
 	{
-		EMapTestingMode TestingMode = SpatialWorldSettings->TestingSettings.TestingMode;
-		if (TestingMode != EMapTestingMode::UseCurrentSettings)
+		FString SnapshotForMap = ASpatialFunctionalTest::GetTakenSnapshotPath(World);
```

Comment:
Were you able to validate that the snapshot tests still pass? Their run steps are a little funky - need to run them twice.

Reply:
I wasn't aware of the but the good news tests pass on the 2nd run. I think this needs to be clearer for those running the tests that are not familiar with them. Maybe pre-fix the test title with RUN_TWICE ??? and a note in USER_MANUAL.md?